packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,630 @@
+          responseCode.start <= statusCode &&
+          responseCode.end >= statusCode
+        ) {
+          responseExamples.set(
+            statusCode,
+            diagnostics.pipe(handleHttpResponse(response, example.responses[code], relativePath))
+          );
+          found = true;
+          break;
+        }
+      }
+      if (!found) {
+        addExampleValueNoMappingDignostic(
+          diagnostics,
+          { [code]: example.responses[code] },
+          relativePath
+        );
+      }
+    }
+  }
+  return diagnostics.wrap(responseExamples);
+}
+
+function handleHttpResponse(
+  response: SdkHttpResponse,
+  example: any,
+  relativePath: string
+): [SdkHttpResponseExample, readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+  const responseExample = {
+    response,
+    headers: [],
+  } as SdkHttpResponseExample;
+  if (typeof example === "object") {
+    for (const name of Object.keys(example)) {
+      if (name === "description") {
+        continue;
+      } else if (name === "body") {
+        if (response.type) {
+          responseExample.value = diagnostics.pipe(
+            getSdkTypeExample(response.type, example.body, relativePath)
+          );
+        } else {
+          addExampleValueNoMappingDignostic(diagnostics, { body: example.body }, relativePath);
+        }
+      } else if (name === "headers") {
+        for (const subName of Object.keys(example.headers)) {
+          const header = response.headers.find((p) => p.serializedName === subName);
+          if (header) {
+            const value = diagnostics.pipe(
+              getSdkTypeExample(header.type, example[name][subName], relativePath)
+            );
+            if (value) {
+              responseExample.headers.push({
+                header,
+                value,
+              });
+            }
+          } else {
+            addExampleValueNoMappingDignostic(
+              diagnostics,
+              { [subName]: example[name][subName] },
+              relativePath
+            );
+          }
+        }
+      } else {
+        addExampleValueNoMappingDignostic(diagnostics, { [name]: example[name] }, relativePath);
+      }
+    }
+  }
+  return diagnostics.wrap(responseExample);
+}
+
+function getSdkTypeExample(
+  type: SdkType | SdkModelPropertyType,
+  example: any,
+  relativePath: string
+): [SdkTypeExample | undefined, readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+
+  if (isSdkIntKind(type.kind) || isSdkFloatKind(type.kind)) {
+    return getSdkBaseTypeExample("number", type as SdkType, example, relativePath);
+  } else if (type.kind === "string" || type.kind === "bytes") {
+    return getSdkBaseTypeExample("string", type as SdkType, example, relativePath);
+  } else if (type.kind === "boolean") {
+    return getSdkBaseTypeExample("boolean", type as SdkType, example, relativePath);
+  } else if (
+    type.kind === "password" ||
```

Comment:
since @ArcturusZhang is working on changing the kinds here, I would recommend going based off of enums and typechecking functions instead of directly listing them out. Just need to make sure that these changes are consistent with the changes that @ArcturusZhang is introducing

Reply:
i'll update the logic after his pr merged.